### PR TITLE
componentconfig: move kube proxy defaulting into the exetrnal api

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: pkg/apis/componentconfig/v1alpha1/defaults.go [hunk 1]
```diff
@@ -40,8 +40,21 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
...
 				obj.OOMScoreAdj = &temp
 			}
+			if obj.ResourceContainer == "" {
+				obj.ResourceContainer = "/kube-proxy"
+			}
 			if obj.IPTablesSyncPeriod.Duration == 0 {
-				obj.IPTablesSyncPeriod = unversioned.Duration{5 * time.Second}
+				obj.IPTablesSyncPeriod = unversioned.Duration{30 * time.Second}
+			}
+			zero := unversioned.Duration{}
```
Comment (thockin) [inline pkg/apis/componentconfig/v1alpha1/defaults.go]: shouldn't we have a const in unversioned `ZeroDuration` or something?

**Comment by thockin** [comment]
overall my biggest grump with this is that these types and logic are way far away from where they are used.  We really need the idea of a set of versioned, convertible types to be librarified.

**Comment by mikedanese** [comment]
@thockin we can create an config API group per component and place each config closer to the components code. What do you think of that?

**Comment by thockin** [comment]
NB I am not blocking this PR - already LGTM'ed

**Pull request merged**